app/src/main/java/com/sma2/sma2/SignalRecording/MovementRecorder.java

```diff
@@ -0,0 +1,308 @@
+package com.sma2.sma2.SignalRecording;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+
+import java.util.Arrays;
+
+public class MovementRecorder implements SensorEventListener {
+    private final String TAG = "MovementRecorder";
+    private SensorManager mSensorManager;
+    private Sensor mAccelerometer;
+    private Sensor mGyroscope;
+    private Sensor mMagnetometer;
+    private Sensor mOrientation;
+    private int mSamplingFrequency_us = SensorManager.SENSOR_DELAY_NORMAL;
+    private CSVFileWriter mCSVFileWriter;
+
+    private CombinedSensorDataFrame combinedSensorDataFrame = null;
+    private static boolean mEnableLogging = false;
+    private final int SYNC_ACCURACY_NS = 200000; // corresponds to maximum difference of 200us in timestamps
+
+    private String[] SENSOR_INFO_HEADER =  {"Name","Vendor","MaxRange","Resolution","Sampling Distance [us]"};
+
+    private String[] SENSOR_DATA_HEADER =  {"Timestamp [ns]",
+            "aX [m/s^2]","aY [m/s^2]","aZ [m/s^2]",
+            "gX [rad/s]","gY [rad/s]","gZ [rad/s]",
+            "mX [uT]","mY [uT]","mZ [uT]",
+            "r0 [a.u.]","r1 [a.u.]","r2 [a.u.]","r3 [a.u.]"};
+
+    public MovementRecorder(Context context, int samplingFrequency_us) {
+        mSamplingFrequency_us = samplingFrequency_us;
+        mSensorManager = (SensorManager) context.getSystemService(context.SENSOR_SERVICE);
+        mAccelerometer = mSensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        mGyroscope = mSensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE);
+        mMagnetometer = mSensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+        mOrientation = mSensorManager.getDefaultSensor(Sensor.TYPE_ROTATION_VECTOR);
+
+        mCSVFileWriter = new CSVFileWriter();
+
+        // generate header information
+        mCSVFileWriter.writeData(SENSOR_INFO_HEADER);
+        if (mAccelerometer != null) {
+            mCSVFileWriter.writeData(new SensorInformation(mAccelerometer,mSamplingFrequency_us).toStringArray());
+        }
+        if (mGyroscope != null) {
+            mCSVFileWriter.writeData(new SensorInformation(mGyroscope,mSamplingFrequency_us).toStringArray());
+        }
+        if (mMagnetometer != null) {
+            mCSVFileWriter.writeData(new SensorInformation(mMagnetometer,mSamplingFrequency_us).toStringArray());
+        }
+        if (mOrientation != null) {
+            mCSVFileWriter.writeData(new SensorInformation(mOrientation,mSamplingFrequency_us).toStringArray());
+        }
+        mCSVFileWriter.writeData(SENSOR_DATA_HEADER);
+
+    }
+
+    public void setSamplingFrequency(int samplingFrequencyMicroSeconds) {
+        mSamplingFrequency_us = samplingFrequencyMicroSeconds;
+    }
+
+    public void registerListeners() {
+        if (mAccelerometer != null) {
```

Comment:
I guess this is fine for now, but can't we handle all these if statements in a loop